pkg/operator/ceph/nfs/controller.go

```diff
@@ -17,214 +17,277 @@ limitations under the License.
-	nfses, err := c.context.RookClientset.CephV1().CephNFSes(c.namespace).List(metav1.ListOptions{})
-	if err != nil {
-		logger.Errorf("failed to retrieve NFSes to update the ceph version. %v", err)
-		return
-	}
-	for _, nfs := range nfses.Items {
-		logger.Infof("updating the ceph version for nfs %q to %q", nfs.Name, c.clusterSpec.CephVersion.Image)
-		err := c.upCephNFS(nfs, 0)
+}
+
+func (r *ReconcileCephNFS) reconcileCreateCephNFS(cephNFS *cephv1.CephNFS) (reconcile.Result, error) {
+	if r.cephClusterSpec.External.Enable {
+		_, err := opcontroller.ValidateCephVersionsBetweenLocalAndExternalClusters(r.context, cephNFS.Namespace, r.clusterInfo.CephVersion)
 		if err != nil {
-			logger.Errorf("failed to update nfs %q. %v", nfs.Name, err)
-		} else {
-			logger.Infof("updated nfs %q to ceph version %q", nfs.Name, c.clusterSpec.CephVersion.Image)
+			// This handles the case where the operator is running, the external cluster has been upgraded and a CR creation is called
+			// If that's a major version upgrade we fail, if it's a minor version, we continue, it's not ideal but not critical
+			return reconcile.Result{}, errors.Wrapf(err, "refusing to run new crd")
 		}
 	}
-}
 
-func nfsChanged(oldNFS, newNFS cephv1.NFSGaneshaSpec) bool {
-	if oldNFS.Server.Active != newNFS.Server.Active {
-		return true
+	err := r.upCephNFS(cephNFS, 0)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to create ceph nfs %q", cephNFS.Name)
 	}
-	return false
-}
-
-func (c *CephNFSController) acquireOrchestrationLock() {
-	logger.Debugf("Acquiring lock for nfs orchestration")
-	c.orchestrationMutex.Lock()
-	logger.Debugf("Acquired lock for nfs orchestration")
-}
-
-func (c *CephNFSController) releaseOrchestrationLock() {
-	c.orchestrationMutex.Unlock()
-	logger.Debugf("Released lock for nfs orchestration")
-}
 
-func ownerRefs(nfs cephv1.CephNFS) []metav1.OwnerReference {
-	// Set the filesystem CR as the owner
-	return []metav1.OwnerReference{{
-		APIVersion: fmt.Sprintf("%s/%s", CephNFSResource.Group, CephNFSResource.Version),
-		Kind:       CephNFSResource.Kind,
-		Name:       nfs.Name,
-		UID:        nfs.UID,
-	}}
+	return reconcile.Result{}, nil
 }
 
-func updateCephNFSStatus(name, namespace, status string, context *clusterd.Context) {
-	updatedCephNFS, err := context.RookClientset.CephV1().CephNFSes(namespace).Get(name, metav1.GetOptions{})
-	if err != nil {
-		logger.Errorf("Unable to update the cephNFS %s status %v", updatedCephNFS.GetName(), err)
-		return
-	}
-	if updatedCephNFS.Status == nil {
-		updatedCephNFS.Status = &cephv1.Status{}
-	} else if updatedCephNFS.Status.Phase == status {
-		return
-	}
-	updatedCephNFS.Status.Phase = status
-	_, err = context.RookClientset.CephV1().CephNFSes(updatedCephNFS.Namespace).Update(updatedCephNFS)
-	if err != nil {
-		logger.Errorf("Unable to update the cephNFS %s status %v", updatedCephNFS.GetName(), err)
-		return
+// func (c *CephNFSController) onUpdate(oldObj, newObj interface{}) {
```

Comment:
this dead code can be removed?

Reply:
I just needed to add support for scale up/down :)